### ISSUE TITLE
Tray icon looks bad with Qt 5.4.0

### ISSUE THREAD

**Issue by zetok** [qTox/qTox#968]
![qt5.4.0 a shit](https://cloud.githubusercontent.com/assets/3148759/5572411/e4850a80-8f95-11e4-9401-35b01bc490c3.png)

Was fine with Qt 5.3.2

**Comment by agilob** [comment]
Can't reproduce with 5.4.0-1

**Comment by zetok** [comment]
Have you tried with i3wm?

**Comment by suhr** [comment]
Try using Qt 5.5.0.
PS: actually had this bug before with lesser qt version. May be qt bug.

**Comment by agilob** [comment]
With KDE 4.14

**Comment by stqism** [comment]
@suhr Isn't it a tad early for 5.5.0?

**Comment by apprb** [comment]
![png](https://cloud.githubusercontent.com/assets/6405341/5585440/6c36a79a-90d6-11e4-8952-1c7e9628766d.png)
Qt 5.4.0-1 under awesome: bug is reproducible.

**Comment by ArchangeGabriel** [comment]
No problem here with QT 5.4 under Plasma 5.1.2 with KF 5.5.

**Comment by nbvcxz** [comment]
"bad" you probably mean white background of the icon. Confirmed - reproduced with Arch Linux 64 lxpanel, qt 5.4.0

**Comment by nurupo** [comment]
Sounds related to [QTBUG-35832](https://bugreports.qt.io/browse/QTBUG-35832), which should have been [partially fixed in Qt 5.4.0](https://qt.gitorious.org/qt/qtbase/source/34ce66cd89ea1c618d8f63dd2d9b95aed0a81b11:dist/changes-5.4.0#L435), but it seem to have broken things for @zetok instead of fixing them : )

There is also another tray-related issue that is open [QTBUG-31762](https://bugreports.qt.io/browse/QTBUG-31762). They all are DE-related.

> Apparently the latest version of Ubuntu Gnome has also broken everything as well (a much lower priority for us).

So just get patient and wait till it's fixed in some future version.

**Comment by tux3** [comment]
Is this still a problem with the new systray code and an up to date Qt version?

**Comment by agilob** [comment]
@apprb could you check what's the different between other icons with transparent background and qtox icon?

Find and icon under /usr/share/application/...  and check its type with `file /path/to/a/file.extenstion`

**SOME** qtox icons are RGBA (colors + transparency), but most of the icons are just RGB (just colors, transparent is white).

```
➜ file img/icons/qtox-*          
img/icons/qtox-128x128.png: PNG image data, 128 x 128, 8-bit/color RGBA, non-interlaced
img/icons/qtox-14x14.png:   PNG image data, 14 x 14, 8-bit colormap, non-interlaced
img/icons/qtox-16x16.png:   PNG image data, 16 x 16, 8-bit colormap, non-interlaced
img/icons/qtox-192x192.png: PNG image data, 192 x 192, 8-bit/color RGBA, non-interlaced
img/icons/qtox-22x22.png:   PNG image data, 22 x 22, 8-bit colormap, non-interlaced
img/icons/qtox-24x24.png:   PNG image data, 24 x 24, 8-bit colormap, non-interlaced
img/icons/qtox-256x256.png: PNG image data, 256 x 256, 8-bit/color RGBA, non-interlaced
img/icons/qtox-32x32.png:   PNG image data, 32 x 32, 8-bit colormap, non-interlaced
img/icons/qtox-36x36.png:   PNG image data, 36 x 36, 8-bit colormap, non-interlaced
img/icons/qtox-48x48.png:   PNG image data, 48 x 48, 8-bit colormap, non-interlaced
img/icons/qtox-512x512.png: PNG image data, 512 x 512, 8-bit/color RGBA, non-interlaced
img/icons/qtox-64x64.png:   PNG image data, 64 x 64, 8-bit colormap, non-interlaced
img/icons/qtox-72x72.png:   PNG image data, 72 x 72, 8-bit colormap, non-interlaced
img/icons/qtox-96x96.png:   PNG image data, 96 x 96, 8-bit colormap, non-interlaced
```

I guess converting it to RGBA will fix the issue. Although, none of those commands worked:

```
convert  xc:transparent -channel RGBA *.png
for i in *.png; do convert $i -transparent white -channel RGBA $(basename $i .png).png; done
for i in *.png; do convert $i xc:transparent -channel RGBA $(basename $i .png).png; done
```

The folder img/icons contains .icns files, so someone need to import it to inkscape and export icons with transparency RGBA.

**Comment by tux3** [comment]
IIRC the .icns is just for OS X, where it's not an issue.

**Comment by agilob** [comment]
Ok, the icons were made by @dubslow in inkscape, I never used it, so I assumed the .icns file are inkscape file format. In this case we need inkscape project. Is that .svg file? can it be imported?

**Comment by tux3** [comment]
I don't think there are many tools that can open .icns files outside of OS X, but there's a icns2png and png2icns tool for converting. It should be in package icnsutils on linux.

I don't know about the inkscape projects, but I'm pretty sure inkscape can open .svg files.

**Comment by zetok** [comment]
Looks like it was fixed in Qt 5.5, only to be replaced by even worse bug → #1820
